### PR TITLE
🍰 Feature/models

### PR DESCRIPTION
**Pull request by memhard**
## What changed?
add models + change path/logic for migrations + add auto-migrations     

## Issue ticket(s) number and link     
- Resolves No link 
     
## Checklist before requesting a review     
- [x] I have performed a self-review of my code       
- [x] Code is functional on my machine(Windows 11/Ubuntu22.04 LTS)

### CODE REVIEW

File: src/db/migrations/env.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 from alembic import context
 
-from src.database import Base
```
Comment (Olderestin) [inline src/db/migrations/env.py:8]: You delete Base import, but were do u get target_metadata from?

File: src/db/auto_migrate.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from alembic.config import Config
+from alembic import command
+
+alembic_cfg = Config("./alembic.ini")
```
Comment (Olderestin) [inline src/db/auto_migrate.py]: Should we use Path there?

File: src/db/migrations/env.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
...
+from models import *
```
Comment (Olderestin) [inline src/db/migrations/env.py]: Why do we need it here?

File: src/db/models/bascket.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+from sqlalchemy import Column, DateTime, Integer, String, JSON, ForeignKey, Numeric
+from sqlalchemy.orm import Relationship
+from db.conection import Base
+
+class Bascet(Base):
+    __tablename__ = "bascet"
+    
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
```
Comment (Olderestin) [inline src/db/models/bascket.py]: Is it one-to-one relationship with user?

File: src/db/models/bascket.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    smartphone_id = Column(Integer, ForeignKey('smartphones.id'), nullable=False)
+    smartphones = Column(JSON, nullable=False)
```
Comment (Olderestin) [inline src/db/models/bascket.py]: Why we need 2 similar colums for smartphones?

File: src/db/models/bascket.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    created_at = Column(DateTime, default=datetime.utcnow)
```
Comment (Olderestin) [inline src/db/models/bascket.py]: It will be created with user registration, so why do we need datetime here?

File: src/db/models/bascket.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    updated_at = Column(DateTime, onupdate=datetime.utcnow)
```
Comment (Olderestin) [inline src/db/models/bascket.py]: Why do we need updated_at column?

File: src/db/models/user.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class User(Base):
+    __tablename__ = "users"
+    
+    id = Column(Integer, primary_key=True)
+    username = Column(String(50), unique=True)
+    surname = Column(String(50))
+    email = Column(String(120), unique=True)
+    _password = Column("password", String(120), nullable=False)
```
Comment (Olderestin) [inline src/db/models/user.py]: As a suggestion, we can use the password hash column.

**Review by Olderestin — COMMENTED** [review]
Questions and suggestions.

File: src/db/models/bascket.py (hunk 5 above)
Comment (Olderestin) [reply src/db/models/bascket.py]: 1. If we use smartphone_id, so we need to make many-to-many relationship, as i know, we need to create association table for it.
2. If we use column smartphones as JSON field, we doesn't need smartphone_id column.

But if u want to use relationship, we should choose first variant.

File: src/db/models/bascket.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    
+    user = Relationship("User", back_populates="bascet", uselist=False)
+    smartphone = Relationship("Smartphone", back_populates="bascet")
```
Comment (Olderestin) [inline src/db/models/bascket.py]: I think we should use baskets as back_populates.

File: src/db/models/smartphone.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    photo_urls = Column(JSON, nullable=False)
+    cores_number = Column(Integer, nullable=False)
+    build_in_memory = Column(Integer, nullable=False)
+    ram = Column(Integer, nullable=False)
+    diagonal = Column(Integer, nullable=False)
+    refresh_rate = Column(Integer, nullable=False)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, onupdate=datetime.utcnow)
+    
+    user = Relationship("User", back_populates="smartphones")
```
Comment (Olderestin) [inline src/db/models/smartphone.py]: If we use many-to-many relationship with Basket table, so we need to add Relationship for it here using this parameter:
- secondary=basket_smartphone_association

File: src/db/auto_migrate.py (hunk 2 above)
Comment (memhard) [reply src/db/auto_migrate.py]: fixed

File: src/db/models/bascket.py (hunk 5 above)
Comment (memhard) [reply src/db/models/bascket.py]: associations created

File: src/db/models/bascket.py (hunk 9 above)
Comment (memhard) [reply src/db/models/bascket.py]: renamed

File: src/db/models/smartphone.py (hunk 10 above)
Comment (memhard) [reply src/db/models/smartphone.py]: added

**Review by memhard — COMMENTED** [review]
tasks done

File: src/db/models/basket.py [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+from sqlalchemy.orm import Relationship
+from db.conection import Base
+
+class Bascet(Base):
+    __tablename__ = "baskets"
+    
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
+    smartphone_id = Column(Integer, ForeignKey('smartphones.id'), nullable=False)
+    smartphones = Column(JSON, nullable=False)
```
Comment (Olderestin) [inline src/db/models/basket.py]: If we create a basket_smartphone_association, so why do we need smartphones column?

**Review by Olderestin — COMMENTED** [review]
Answer one question and merge.

File: src/db/models/basket.py (hunk 11 above)
Comment (memhard) [reply src/db/models/basket.py]: removed

**Pull request merged**